docs/content/doc/developers/guidelines-frontend.md

```diff
@@ -40,25 +40,30 @@ We recommend [Google HTML/CSS Style Guide](https://google.github.io/styleguide/h
 1. Every feature (Fomantic-UI/jQuery module) should be put in separate files/directories.
 2. HTML ids and classes should use kebab-case.
 3. HTML ids and classes used in JavaScript should be unique for the whole project, and should contain 2-3 feature related keywords. We recommend to use the `js-` prefix for classes that are only used in JavaScript.
-4. jQuery events across different features should use their own namespaces.
-5. CSS styling for classes provided by frameworks should not be overwritten. Always use new class-names to overwrite framework styles. We recommend to use the `us-` prefix for user defined styles.  
+4. jQuery events across different features could use their own namespaces if there are potential conflicts.
+5. CSS styling for classes provided by frameworks should not be overwritten. Always use new class-names with 2-3 feature related keywords to overwrite framework styles.  
 6. The backend can pass complex data to the frontend by using `ctx.PageData["myModuleData"] = map[]{}`
 7. Simple pages and SEO-related pages use Go HTML Template render to generate static Fomantic-UI HTML output. Complex pages can use Vue2 (or Vue3 in future).
 
 
 ### Framework Usage
 
-Mixing different frameworks together is highly discouraged. A JavaScript module should follow one major framework and follow the framework's best practice.
+Mixing different frameworks together is discouraged, it makes the code difficult to be maintained.
+A JavaScript module should follow one major framework and follow the framework's best practice.
 
 Recommended implementations:
 * Vue + Native
 * Fomantic-UI (jQuery)
 * Native only
```

Comment:
I am missing a statement about Vue + Go templates.
Native sounds like [Vanilla JS](http://vanilla-js.com/) to me.

Reply:
Updated all `Native` to `Vanilla JS`.

Go template is used everywhere and at the moment, it only provides basic HTML layout and all other frameworks are working on top of Go templates in Gitea, it's difficult to describe their relations. At the moment there are Go template + Vue, Go template + jQuery, Go template + Fomantic UI, Go template + Vanilla JS.

 Maybe we can leave the Go template until next milestone.